app/scripts/translator-screenshots/lib.js

```diff
@@ -0,0 +1,473 @@
+              matches.push({
+                tagName: el.tagName.toLowerCase(),
+                directText: directText,
+                placeholderText: placeholderText,
+                matchType:
+                  placeholderText === searchText ? "placeholder" : "text",
+                area: rect.width * rect.height,
+                xpath: getXPath(el),
+              });
+            }
+          }
+        }
+
+        return matches;
+      },
+      [text],
+    );
+
+    if (candidates.length === 0) {
+      return null;
+    }
+
+    // Pick the best candidate (prefer direct text match, then placeholder, then smallest area)
+    const bestCandidate = candidates.reduce((best, current) => {
+      // Priority: direct text > placeholder > full text
+      const bestDirectMatch = best.directText === text;
+      const currentDirectMatch = current.directText === text;
+      const bestPlaceholderMatch = best.placeholderText === text;
+      const currentPlaceholderMatch = current.placeholderText === text;
+
+      // Direct text matches are highest priority
+      if (currentDirectMatch && !bestDirectMatch) {
+        return current;
+      }
+      if (bestDirectMatch && !currentDirectMatch) {
+        return best;
+      }
+
+      // If both/neither are direct matches, prefer placeholder matches
+      if (currentPlaceholderMatch && !bestPlaceholderMatch) {
+        return current;
+      }
+      if (bestPlaceholderMatch && !currentPlaceholderMatch) {
+        return best;
+      }
+
+      // If same match type, prefer smaller area
+      return current.area < best.area ? current : best;
+    });
+
+    try {
+      const element = await this.page.locator(`xpath=${bestCandidate.xpath}`);
+      const count = await element.count();
+      if (count > 0) {
+        return element.first();
+      }
+    } catch (e) {
+      // XPath lookup failed
+    }
+
+    return null;
+  }
+
+  /**
+   * Highlight an element with red border and padding
+   */
+  async _highlightElement(element) {
+    await element.evaluate((el) => {
+      el.style.border = "2px solid red";
+      el.style.borderRadius = "4px";
+      el.style.boxShadow = "0 0 8px rgba(255, 0, 0, 0.4)";
+      el.style.padding = "4px";
+      el.style.position = "relative";
+      el.style.zIndex = "9999";
+      el.dataset.playwrightHighlight = "true";
+    });
+  }
+
+  /**
+   * Clear all highlights from the page
+   */
+  async _clearAllHighlights() {
+    await this.page.evaluate(() => {
+      const highlightedElements = document.querySelectorAll(
+        "[data-playwright-highlight]",
+      );
+      highlightedElements.forEach((el) => {
+        el.style.border = "";
+        el.style.borderRadius = "";
+        el.style.boxShadow = "";
+        el.style.padding = "";
+        el.style.position = "";
+        el.style.zIndex = "";
+        delete el.dataset.playwrightHighlight;
+      });
+
+      // Brute force cleanup
```

Comment:
just curious if there are often elements that are missed using the `data-` selector?

Reply:
If I remember correctly, I added this in because the earlier cleanup wasn't working...let me debug that a bit more.

Reply:
ok it's not actually being invoked, I'll delete it.